library/src/armory/model/object_detection/yolov4_object_detector.py

```diff
@@ -0,0 +1,246 @@
+        predictions_accessor: Optional[Accessor] = None,
+    ):
+        """
+        Initializes the model wrapper.
+
+        Args:
+            name: Name of the model.
+            model: YOLOv4 model being wrapped.
+            inputs_accessor: Optional, data accessor used to obtain low-level
+                image data from the highly-structured image inputs contained in
+                object detection batches. Defaults to an accessor compatible
+                with typical YOLOv4 models.
+            predictions_accessor: Optional, data accessor used to update the
+                object detection predictions in the batch. Defaults to an
+                accessor compatible with typical YOLOv4 models.
+        """
+        super().__init__(
+            name=name,
+            model=model,
+            inputs_accessor=(
+                inputs_accessor
+                or Images.as_torch(
+                    dim=ImageDimensions.CHW,
+                    scale=Scale(dtype=DataType.FLOAT, max=1.0),
+                    dtype=torch.float32,
+                )
+            ),
+            predictions_accessor=(
+                predictions_accessor or BoundingBoxes.as_torch(format=BBoxFormat.XYXY)
+            ),
+        )
+
+    def predict(self, batch: Batch):
+        """
+        Invokes the wrapped model using the image inputs in the given batch and
+        updates the object detection predictions in the batch.
+
+        Non-maximum suppression processing is applied to the model's outputs
+        before the batch predictions are updated.
+
+        Args:
+            batch: Object detection batch
+        """
+        self.eval()
+        inputs = self.inputs_accessor.get(batch.inputs)
+        _, _, h, w = inputs.shape  # (N, C, H, W)
+        outputs = self(inputs)
+        outputs = _post_processing(outputs)
+        if all(not sublist for sublist in outputs):
+            outputs_dict = [
+                {
+                    "boxes": np.array([]),
+                    "labels": np.array([]),
+                    "scores": np.array([]),
+                }
+                for output in outputs
+            ]
+            self.predictions_accessor.set(batch.predictions, outputs_dict)
+        else:
+            outputs_dict = [
+                {
+                    "boxes": np.array(
+                        [
+                            [output[0] * w, output[1] * h, output[2] * w, output[3] * h]
+                            for output in outputs[0][:4]
+                        ],
+                        dtype=np.float32,
+                    ),
+                    "labels": np.array(
+                        [int(output[5]) for output in outputs[0]], dtype=np.int64
+                    ),
+                    "scores": np.array(
+                        [float(output[4]) for output in outputs[0]], dtype=np.float32
+                    ),
+                }
+                for output in outputs
```

Comment:
@jxcronay What is the actual shape of YOLOv4 output? This looping logic here doesn't seem correct in that it is looping through `output` to populate the list of detections for each image in the batch, but each image is just looping through `outputs[0]`.

Reply:
Updates pushed.